src/core/lis-modal-element.ts

```diff
@@ -0,0 +1,137 @@
+import {LitElement, html, css} from 'lit';
+import {customElement, property} from 'lit/decorators.js';
+import {unsafeHTML} from 'lit/directives/unsafe-html.js';
+
+
+/**
+ * @htmlElement `<lis-modal-element>`
+ *
+ * A Web Component that provides a generic modal element.
+ *
+ * @example
+ * The modal element's
+ * {@link name | `name`}, {@link heading | `heading`}, and
+ * {@link content | `content`} attributes/properties can be set via HTML or
+ * javascript.
+ *
+ * For example:
+ * ```html
+ * <!-- set the name, heading and content via HTML -->
+ * <lis-modal-element
+ *   name="modal-test"
+ *   heading="Test Modal">
+ *     <p>Some HTML or text to be rendered</p>
+ * </lis-modal-element>
+ * ```
+ *
+ * @example
+ * In the example below, the lis-simple-table-element web component
+ * is rendered within the lis-modal-element.
+ *
+ * The attributes/properties for lis-simple-table-element are set below
+ * in javascript. Please see the documentation for lis-simple-table-element
+ * for more information.
+ *
+ * ```html
+ * <lis-modal-element 
+ *   name="modal-test" 
```

Comment:
Change name to modal-id

Reply:
I think this has to be `modalId`. I don't think it maps kebab-case to camelCase?

Reply:
Oh OK that sounds good.